## Module 3 Coding Exercises

In [7]:
# 2 start your code by importing pandas
import pandas as pd

In [8]:
# 3 create 2 series from list
author = pd.Series(['E.L. James', 'Suzanne Collins', 'J.K Rowling', 'Dan Brown'])
reviews = pd.Series([410, 211, 314, 178])

# 4 display both series using print function
print('Author')
print(author)
print('\n')
print('Reviews')
print(reviews)


Author
0         E.L. James
1    Suzanne Collins
2        J.K Rowling
3          Dan Brown
dtype: object


Reviews
0    410
1    211
2    314
3    178
dtype: int64


In [9]:
# 6 create dataframe from the csv file
df = pd.read_csv('bestsellers.csv')

df

Name              Author  \
0                        10-Day Green Smoothie Cleanse            JJ Smith   
1                                                  NaN                 NaN   
2                                    11/22/63: A Novel        Stephen King   
3              12 Rules for Life: An Antidote to Chaos  Jordan B. Peterson   
4                               1984 (Signet Classics)       George Orwell   
..                                                 ...                 ...   
370  Winter of the World: Book Two of the Century T...         Ken Follett   
371  Women Food and God: An Unexpected Path to Almo...         Geneen Roth   
372                                             Wonder       R. J. Palacio   
373       Wrecking Ball (Diary of a Wimpy Kid Book 14)         Jeff Kinney   
374  You Are a Badass: How to Stop Doubting Your Gr...         Jen Sincero   

     User Rating  Reviews  Price    Year        Genre         Type  
0            4.7  17350.0    8.0  2016.0  Non Fiction  Non Fiction  
1            NaN      NaN    NaN     NaN          NaN          NaN  
2            4.6   2052.0   22.0  2011.0      Fiction      Fiction  
3            4.7  18979.0   15.0  2018.0  Non Fiction  Non Fiction  
4            4.7  21424.0    6.0  2017.0      Fiction      Fiction  
..           ...      ...    ...     ...          ...          ...  
370          4.5  10760.0   15.0  2012.0      Fiction      Fiction  
371          4.2   1302.0   11.0  2010.0  Non Fiction  Non Fiction  
372          4.8  21625.0    9.0  2017.0      Fiction      Fiction  
373          4.9   9413.0    8.0  2019.0      Fiction      Fiction  
374          4.7  14331.0    8.0  2019.0  Non Fiction  Non Fiction  

[375 rows x 8 columns]

In [10]:
# 7 display the first 5 rows
df.head()

Name              Author  User Rating  \
0            10-Day Green Smoothie Cleanse            JJ Smith          4.7   
1                                      NaN                 NaN          NaN   
2                        11/22/63: A Novel        Stephen King          4.6   
3  12 Rules for Life: An Antidote to Chaos  Jordan B. Peterson          4.7   
4                   1984 (Signet Classics)       George Orwell          4.7   

   Reviews  Price    Year        Genre         Type  
0  17350.0    8.0  2016.0  Non Fiction  Non Fiction  
1      NaN    NaN     NaN          NaN          NaN  
2   2052.0   22.0  2011.0      Fiction      Fiction  
3  18979.0   15.0  2018.0  Non Fiction  Non Fiction  
4  21424.0    6.0  2017.0      Fiction      Fiction

In [11]:
# 8 information of file
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 375 entries, 0 to 374
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Name         347 non-null    object 
 1   Author       347 non-null    object 
 2   User Rating  347 non-null    float64
 3   Reviews      347 non-null    float64
 4   Price        347 non-null    float64
 5   Year         347 non-null    float64
 6   Genre        347 non-null    object 
 7   Type         347 non-null    object 
dtypes: float64(4), object(4)
memory usage: 23.6+ KB


In [12]:
# 8 shape of file in terms of number of cloumns and rows
df.shape

(375, 8)

# ----

# Coding Exercise Continuation 3.8

### Look at the total null values and decide whether to drop them or not

In [25]:
# we need to find out the shape of our data
df.shape

(375, 8)

We have 375 data entry in our dataframe

In [27]:
# we need to find how many null values arethre in our dataframe
df.isnull().sum()

Name           28
Author         28
User Rating    28
Reviews        28
Price          28
Year           28
Genre          28
Type           28
dtype: int64

Using .isnull method, I found out that there is 28 null values in our dataframe

In [28]:
# findout out the null values if it correlates with our counted data
df.describe()

User Rating       Reviews       Price         Year
count   347.000000    347.000000  347.000000   347.000000
mean      4.607493   9709.674352   13.095101  2013.786744
std       0.227582  10831.206514   10.093187     3.360868
min       3.300000     37.000000    0.000000  2009.000000
25%       4.500000   3384.500000    8.000000  2011.000000
50%       4.600000   6310.000000   12.000000  2014.000000
75%       4.800000  11259.500000   16.000000  2017.000000
max       4.900000  87841.000000  105.000000  2019.000000

In our count, we only have 347 of 375 data entry, which the 28 null values describe by  .isnull method is relevant.

In [29]:
# dropping the null values for a cleaner data
df = df.dropna()  # remove the null values in dataframe
df.isnull().sum()  # check if theres still null values

Name           0
Author         0
User Rating    0
Reviews        0
Price          0
Year           0
Genre          0
Type           0
dtype: int64

we removed the data which is null values and therefore achieve 0 values

### Find the basic info and th efirst five rows to see my cleaned dataframe

In [31]:
# information about my dataframe
df.info() 

# display first 5 rows
df.head()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 347 entries, 0 to 374
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Name         347 non-null    object 
 1   Author       347 non-null    object 
 2   User Rating  347 non-null    float64
 3   Reviews      347 non-null    float64
 4   Price        347 non-null    float64
 5   Year         347 non-null    float64
 6   Genre        347 non-null    object 
 7   Type         347 non-null    object 
dtypes: float64(4), object(4)
memory usage: 24.4+ KB


Name  \
0                      10-Day Green Smoothie Cleanse   
2                                  11/22/63: A Novel   
3            12 Rules for Life: An Antidote to Chaos   
4                             1984 (Signet Classics)   
5  5,000 Awesome Facts (About Everything!) (Natio...   

                     Author  User Rating  Reviews  Price    Year        Genre  \
0                  JJ Smith          4.7  17350.0    8.0  2016.0  Non Fiction   
2              Stephen King          4.6   2052.0   22.0  2011.0      Fiction   
3        Jordan B. Peterson          4.7  18979.0   15.0  2018.0  Non Fiction   
4             George Orwell          4.7  21424.0    6.0  2017.0      Fiction   
5  National Geographic Kids          4.8   7665.0   12.0  2019.0  Non Fiction   

          Type  
0  Non Fiction  
2      Fiction  
3  Non Fiction  
4      Fiction  
5  Non Fiction

### We can conclude that the data is cleaner now because of 345 entries where we dropped the null values for easier diagnostic and analysis of our data.